### PR TITLE
Simplify codegen tests

### PR DESCRIPTION
**Pull request by timowest**

### CODE REVIEW

**Comment by Shredder121** [comment]
I think that using `ErrorCollector` would be nice.

File: querydsl-jpa-codegen/src/test/java/com/querydsl/jpa/codegen/HibernateDomainExporterTest.java [hunk 1]
```diff
@@ -13,204 +13,54 @@
...
+        Set<File> files = exporter.getGeneratedFiles();
+        assertFalse(files.isEmpty());
+        for (File file : files) {
+            String path = file.getAbsolutePath().replace(
+                    folder.getRoot().getAbsolutePath(), origRoot.getAbsolutePath());
+            String reference = Files.toString(new File(path), Charsets.UTF_8);
+            String content = Files.toString(file, Charsets.UTF_8);
+            try {
+                assertEquals(file.getPath(), reference, content);
+            } catch (Throwable e) {
```
Comment (Shredder121) [inline querydsl-jpa-codegen/src/test/java/com/querydsl/jpa/codegen/HibernateDomainExporterTest.java]: I would write it like this, it does away with the try-catch:
```java
errors.checkThat(file.getPath(), content, is(equalTo(reference)));
```

**Comment by robertandrewbain** [comment]
Looks good to me.

File: querydsl-jpa-codegen/src/test/java/com/querydsl/jpa/codegen/HibernateDomainExporterTest.java [hunk 2]
```diff
@@ -13,203 +13,50 @@
...
+
+        File origRoot = new File("../querydsl-jpa/target/generated-test-sources/java");
+        Set<File> files = exporter.getGeneratedFiles();
+        assertFalse(files.isEmpty());
+        for (File file : files) {
+            String path = file.getAbsolutePath().replace(
+                    folder.getRoot().getAbsolutePath(), origRoot.getAbsolutePath());
+            String reference = Files.toString(new File(path), Charsets.UTF_8);
+            String content = Files.toString(file, Charsets.UTF_8);
+            errors.checkThat(content, is(equalTo(reference)));
```
Comment (Shredder121) [inline querydsl-jpa-codegen/src/test/java/com/querydsl/jpa/codegen/HibernateDomainExporterTest.java]: Don't you want to keep the `reason` parameter in this?

File: querydsl-jpa-codegen/src/test/java/com/querydsl/jpa/codegen/ant/AntJPADomainExporterTest.java [hunk 3]
```diff
@@ -1,25 +1,43 @@
...
-        assertTrue(new File("target/AntJPADomainExporterTest/com/querydsl/jpa/codegen/ant/QDepartments.java").exists());
+        File origRoot = new File("../querydsl-jpa/target/generated-test-sources/java");
+        Set<File> files = exporter.getGeneratedFiles();
+        assertFalse(files.isEmpty());
+        for (File file : files) {
+            String path = file.getAbsolutePath().replace(
+                    folder.getRoot().getAbsolutePath(), origRoot.getAbsolutePath());
+            String reference = Files.toString(new File(path), Charsets.UTF_8);
+            String content = Files.toString(file, Charsets.UTF_8);
+            assertEquals(file.getPath(), reference, content);
```
Comment (Shredder121) [inline querydsl-jpa-codegen/src/test/java/com/querydsl/jpa/codegen/ant/AntJPADomainExporterTest.java]: You don't want to move them all to hamcrest?

**Comment by Shredder121** [comment]
Could @timowest maybe bump the minimum coverage to 75% on coveralls after this is merged?
Then we at least are anchored and secured.

File: querydsl-jpa-codegen/src/test/java/com/querydsl/jpa/codegen/HibernateDomainExporterTest.java (hunk 2 above)
Comment (timowest) [inline querydsl-jpa-codegen/src/test/java/com/querydsl/jpa/codegen/HibernateDomainExporterTest.java]: how to get the reason in?

Comment (Shredder121) [inline querydsl-jpa-codegen/src/test/java/com/querydsl/jpa/codegen/HibernateDomainExporterTest.java]: You're now using this one.
`public <T> void checkThat(T value, Matcher<T> matcher) ...`
You can also use this one
`public <T> void checkThat(String reason, T value, Matcher<T> matcher) ...`

Is `String path` a readable path of the file?
Or else `file.toString()`?

Comment (Shredder121) [inline querydsl-jpa-codegen/src/test/java/com/querydsl/jpa/codegen/HibernateDomainExporterTest.java]: Oh it seems we're on an older version of junit than I usually am.
I am at `junit-4.12` and use ErrorCollector myself, so thought that the `reason` would make more sense.
You can either bump the junit version (and see which surefire version works) or just leave it like this.

**Comment by timowest** [comment]
Yes, i'll bump the minimum coverage.

**Comment by Shredder121** [comment]
They added some validation regarding `@Rule` and static `TestRule`s/`MethodRule`s.
I am seeing if I can rewrite them as `TestRule`s, since they are a little cleaner.

**Comment by Shredder121** [comment]
I am done, tell me how it looks?

**Comment by timowest** [comment]
Looks good.

**Comment by johnktims** [comment]
Looks good to me.

**Pull request merged**